### PR TITLE
Optimizes the game across the board

### PR DESCRIPTION
**Pull request by ike709**
## Changelog
:cl:
code: Tons of code optimizations across the board, both big and small.
code: Micro-optimized playsound()
code: A ton of other micro-optimizations.
code: Replaced get_hearers_in_view() with hearers() in many places. This may break hearing some messages, hearing some sounds, or seeing some speech bubbles. Please report issues on github.
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
Completely untested, should improve several aspects of saycode performance.

**Review by qwertyquerty — DISMISSED** [review]
Skimmed, saw no issues

**Review by qwertyquerty — DISMISSED** [review]

**Comment by ike709** [comment]
This breaks headset tcomms. Station intercoms still work.

File: code/__HELPERS/game.dm
Comment (ike709) [inline code/__HELPERS/game.dm]:
```suggestion
				. |= hearers(R.canhear_range, get_turf(R))
```
Possible fix, needs testing.

**Comment by park66665** [comment]
The code looks good to me; needs more testing, though

**Comment by ike709** [comment]
Tcomms is fixed.

File: code/modules/mob/mob.dm
Comment (ike709) [inline code/modules/mob/mob.dm]:
```suggestion
	for(var/mob/M as() in hearers(range, T))
```

Comment (ike709) [reply code/modules/mob/mob.dm]: I haven't seen any bugs with visible_message() yet but this will likely prevent them before they crop up.

File: code/modules/mob/mob.dm
Comment (ike709) [inline code/modules/mob/mob.dm]:
```suggestion
	for(var/mob/M as() in hearers(range, get_turf(src)))
```
Ditto.

File: code/modules/mob/mob.dm
Comment (ike709) [inline code/modules/mob/mob.dm]:
```suggestion
	for(var/mob/M as() in hearers(range, get_turf(src)))
```

Comment (ike709) [reply code/modules/mob/mob.dm]: Ditto.

File: code/modules/surgery/organs/vocal_cords.dm
Comment (ike709) [inline code/modules/surgery/organs/vocal_cords.dm]:
```suggestion
	for(var/mob/living/L in hearers(8, get_turf(user)))
```

Comment (ike709) [reply code/modules/surgery/organs/vocal_cords.dm]: Ditto

**Comment by ike709** [comment]
Needs further testing but it *should* be fine.

**Review by park66665 — APPROVED** [review]
I read through changes about three times. Nothing seems to be problem-prone. This PR will (technically) need Bacon's approval, however.

**Comment by ike709** [comment]
> This PR will (technically) need Bacon's approval, however.

Code owners get rekt.

**Comment by 4rchibald** [comment]
@ike709 
So what's the feedback from players so far?
Even since we got this test merged I got constant lags (and from what I gather from OOC others experience them too), I also had some weird graphics issue 2 days ago, never had it before nor after that. 

beebug

Can't tell if its related to this PR, something else or maybe its all on my shitty pc but I feel like lag issues started with test merge so I'm throwing it out there, maybe others will share similar issues

**Pull request merged**
